### PR TITLE
[Coop] ves_icall_InternalInvoke.

### PR DESCRIPTION
**Pull request by jaykrell**
(Part taken from ves_icall_InternalExecute https://github.com/mono/mono/pull/16622.)

### CODE REVIEW

**Comment by jaykrell** [comment]
One relevant test failure.

File: mono/metadata/icall.c [hunk 1]
```diff
@@ -3582,167 +3582,174 @@ ves_icall_RuntimeMethodInfo_GetGenericArguments (MonoReflectionMethodHandle ref_
...
 	 */
-	MonoMethod *m = method->method;
-	MonoMethodSignature *sig = mono_method_signature_internal (m);
-	MonoImage *image;
-	int pcount;
-	void *obj = this_arg;
-
-	*exc = NULL;
+	MonoMethod* m = method->method;
+	MonoMethodSignature* const sig = mono_method_signature_internal (m);
```
Comment (vargaz) [inline mono/metadata/icall.c]: Please use MonoMethod *m, thats the mono (and C) coding convention.

File: mono/metadata/icall.c [hunk 2]
```diff
@@ -3582,167 +3582,174 @@ ves_icall_RuntimeMethodInfo_GetGenericArguments (MonoReflectionMethodHandle ref_
...
+	MonoException* exception = NULL;
+
+	*MONO_HANDLE_REF (exception_out) = NULL;
 
 	if (mono_security_core_clr_enabled () &&
 	    !mono_security_core_clr_ensure_reflection_access_method (m, error)) {
-		mono_error_set_pending_exception (error);
-		return NULL;
+		goto return_null;
 	}
```
Comment (vargaz) [inline mono/metadata/icall.c]: Whats the point of this change ?

Comment (jaykrell) [reply mono/metadata/icall.c]: Single point of return is often useful, but less critical in this case. I can probably reduce the change.

Comment (jaykrell) [reply mono/metadata/icall.c]: I've left it alone for now.

Comment (jaykrell) [reply mono/metadata/icall.c]: By "left it alone" I mean, "I left my change in".

**Comment by jaykrell** [comment]
@monojenkins build failed

**Pull request merged**